Automattic/wp-calypso
# Activity Log: Update to wpcom/v2 endpoint with Activity Streams

This is a significant refactor. The motivating change is the move to the APIv2 endpoint. As this requires a refactor of the data-related parts of Activity Log, `fromApi` has been implemented, and the structure of data used throughout Calypso UI has been changed.

## Notable changes
* Move to the `wpcom/v2` API endpoint.
* Endpoint data is now structured according to [Activity Streams spec](http://w3c.github.io/activitystreams/core/index.html)
* Activity data-layer `fromApi` is implemented. Significantly flattens, transforms and prunes data as needed for UI.
* `ActivityTitle` is removed in favor of API provided text.
* `ActivityIcon` is vastly simplified relying on API provided icon. It should be further simplified as `status` logic moves to the API.
* Activity `object`s have been dropped as it will not be used for now and would need to be rewritten. `object` will be restored and `target` will be added in the future.
* Updates persistence schema for new data structure

## Testing
* Visit https://calypso.live/stats/activity?branch=update/activity-log/api-endpoint-v2
* Verify persistence works correctly (do you see schema validation errors in the console when you refresh?)
* Verify Activity loads and is displayed correctly
* Verify the new endpoint is used (network requests to `https://public-api.wordpress.com/wpcom/v2/sites/$SITE_ID/activity`)

## Follow-up
* Move `status` logic to API
* Rebase and land #17423
